### PR TITLE
[JIT] Add mypy type annotations for JIT

### PR DESCRIPTION
**Pull request by SplitInfinity**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#43862 [JIT] Add mypy type annotations for JIT**

Differential Revision: [D23491151](https://our.internmc.facebook.com/intern/diff/D23491151)

### CODE REVIEW

**Comment by SplitInfinity** [comment]
Note that this PR enables only regular mypy type checks (not strict, that's the next step).

File: tools/pyi/gen_pyi.py [hunk 1]
```diff
@@ -680,6 +680,11 @@ def gen_pyi(declarations_path, out):
 
     write(out, 'torch/_C/__init__.pyi', TORCH_C_TYPE_STUBS, env)
     write(out, 'torch/_C/_VariableFunctions.pyi', TORCH_C_VARIABLE_FUNCTIONS_TYPE_STUBS, env)
+
+    if not os.path.exists('torch/_VF'):
+        os.mkdir('torch/_VF')
```
Comment (rgommers) [inline tools/pyi/gen_pyi.py:685]: Simply adding `_VF.pyi` as a file seems easier, no need for a separate directory. `torch/_C/` is a bit special and not quite the example to follow; `_VF.py` is a regular Python file.

Comment (SplitInfinity) [reply tools/pyi/gen_pyi.py:685]: So, `torch/_VF.py`?

Comment (rgommers) [reply tools/pyi/gen_pyi.py:685]: `torch/_VF.pyi`

File: torch/jit/_trace.py [hunk 2]
```diff
@@ -202,12 +216,6 @@ def verify(model, args, loss_fn=torch.sum, devices=None):
     # should be possible to check if our execution actually obeyed the 'devices'
     # the user provided.
 
-    # TODO: Consider adding a utility function to torch.jit to test
-    # for this case
-    if not isinstance(model, torch._C.CompiledFunction):
-        raise TypeError(
-            "Cannot verify an uncompiled module.  Add @torch.jit.compile to compile it"
-        )
```
Comment (rgommers) [inline torch/jit/_trace.py:210]: This change seems unrelated to adding type annotations. I can't judge if this check is or isn't still needed, but it's a functional change - can you comment on why this is okay?

**Review by rgommers — APPROVED** [review]
Thanks @SplitInfinity, overall this looks good to me, I just have one clarifying question left.

File: torch/jit/_trace.py (hunk 2 above)
Comment (SplitInfinity) [reply torch/jit/_trace.py:210]: I actually don't know if it's okay or not either, but I do know that `torch._C.CompiledFunction` doesn't exist. I searched the entire codebase for `CompiledFunction` and found nothing.

Comment (rgommers) [reply torch/jit/_trace.py:210]: Ah, it's not used at all currently, and the one test for it in `test/test_jit.py` says:
```
@unittest.skip("verify needs to be updated to work with GraphExecutors")
```

Which probably means you want to put back this code and slap a `# type: ignore` on it.

Comment (SplitInfinity) [reply torch/jit/_trace.py:210]: Okay, makes sense.

File: torch/jit/supported_ops.py [hunk 3]
```diff
@@ -249,12 +261,12 @@ def _get_global_builtins():
...
-    magic_methods_rows = '\n'.join(magic_methods_rows)
-    schematized_ops = textwrap.indent(schematized_ops, '\t')
-    schemaless_ops = textwrap.indent(schemaless_ops, '\t')
-    magic_methods_rows = textwrap.indent(magic_methods_rows, '\t')
+    schematized_ops_str = '\n'.join(schematized_ops)
+    schemaless_ops_str = '\n'.join(schemaless_ops)
+    magic_methods_rows_str = '\n'.join(magic_methods_rows)
+    schematized_ops_str = textwrap.indent(schematized_ops_str, '\t')
+    schemaless_ops_str = textwrap.indent(schemaless_ops_str, '\t')
+    magic_methods_rows_str = textwrap.indent(magic_methods_rows_str, '\t')
```
Comment (rgommers) [inline torch/jit/supported_ops.py:269]: You renamed these variables, but didn't use the new names further down. That explains the doc build failure in CI.

**Pull request closed without merging**